python/paddle/jit/dy2static/pir_partial_program.py

```diff
@@ -223,25 +222,68 @@ def __init__(
             in_out_values[0], list
         ), "in_out_values must be tuple with len == 3"
         self.program = program
-        self.x_names = self.convert_name(in_out_values[0])
-        self.param_names = self.convert_name(in_out_values[1])
-        self.out_names = self.convert_name(in_out_values[2])
+        self.x_names = self.convert_to_name(in_out_values[0])
+        self.param_names = self.convert_to_name(in_out_values[1])
+        self.out_names = self.convert_to_name(in_out_values[2])
+        if grad_in_out_values is None:
+            grad_in_out_values = [], [], []
+        self.x_grad_names = self.convert_to_name(grad_in_out_values[0])
+        self.p_grad_names = self.convert_to_name(grad_in_out_values[1])
+        self.o_grad_names = self.convert_to_name(grad_in_out_values[2])
         self.forward_range = forward_range
+        if self.forward_range is None:
+            self.forward_range = (0, len(self.program.global_block().ops))
         self.backward_range = backward_range
+        if self.backward_range is None:
+            self.backward_range = (
+                len(self.program.global_block().ops),
+                len(self.program.global_block().ops),
+            )
         self.has_splited = False
         self.finish_pass = False
-        if self.forward_range is None:
+
+        # Flag operator, indicating the operator between the forward subgraph and the backward subgraph. After self.program is updated by the pass, it is recommended to use the self.update_op_range interface to update the forward_range and backward_range.
+        self.fwd_end_next_op = (
+            self.program.global_block().ops[self.forward_range[1]]
+            if self.forward_range[1] < len(self.program.global_block().ops)
+            else None
+        )
+        self.bwd_start_pre_op = (
+            self.program.global_block().ops[self.backward_range[0] - 1]
+            if self.backward_range[0] - 1 < len(self.program.global_block().ops)
+            else None
+        )
+        self.bwd_end_nex_op = (
```

Comment:
```suggestion
        self.bwd_end_next_op = (
```

这个 OP 的作用是？反向之后还有些 shadow output op 是么？